packages/fe/src/app/components/core/form/widgets/DataSelectWidget.tsx

```diff
@@ -0,0 +1,164 @@
+import { ChangeEvent, FocusEvent } from 'react';
+import {
+  ariaDescribedByIds,
+  labelValue,
+  FormContextType,
+  RJSFSchema,
+  StrictRJSFSchema,
+  WidgetProps,
+} from '@rjsf/utils';
+import {MenuItem, TextField, TextFieldProps} from "@mui/material";
+import {types} from "@app/shared/types";
+import {isQueryableStateListDescription, QueryableStateListDescription} from "@event-engine/descriptions/descriptions";
+import {useApiQuery} from "@frontend/queries/use-api-query";
+import * as jexl from "jexl";
+import AddBrand from "@frontend/app/components/fleet-management/commands/AddBrand";
+import {commands} from "@frontend/app/components/commands";
+
+// Copied from: https://github.com/rjsf-team/react-jsonschema-form/blob/main/packages/material-ui/src/SelectWidget/SelectWidget.tsx
+// and modified to use useApiQuery and turn result into select options
+
+interface ParsedUiOptions {
+  data: QueryableStateListDescription,
+  label: string,
+  value: string,
+  addItemCommand: string | null,
+}
+
+const parseOptions = (options: any): ParsedUiOptions => {
+
+  if(!options.data || typeof options.data !== "string") {
+    throw new Error('DataSelect: no "data" attribute configured!');
+  }
+
+  if(!types[options.data]) {
+    throw new Error(`DataSelect: Unknown type "${options.data}"`);
+  }
+
+  const vo = types[options.data];
```

Comment:
Every type in the Cody Engine like commands, events, queries and value objects have a so called "runtime info" available via type specific registry files.

Here we load the runtime info of the state list type configured via `ui:options`. The runtime info includes information about the query used to fetch the state list form the backend.